src/Microsoft.AspNetCore.Mvc.Core/RequireHttpsAttribute.cs

```diff
@@ -82,6 +88,9 @@ protected virtual void HandleNonHttpsRequest(AuthorizationFilterContext filterCo
                     host = new HostString(host.Host);
                 }
 
+                //i use MvcOption.requireHttpsPermanent value if Permanent parameter is null
+                _permanent = _permanent ?? optionsAccessor.Value.RequireHttpsPermanent;
```

Comment:
I think there's functionality for reloading configuration (i.e. when a config file changes). Whether that also propagates into options (bound to configuration), I'm not sure about.

Also, changing this exact value at runtime makes little sense to me. I just thought it was worth asking :smile:

Either way, I don't think a new field is necessary.